### ISSUE TITLE
Microceph disk add error

### ISSUE THREAD

**Issue by james-luther** [canonical/microceph#79]
When attempting to add disks to a new cluster or an existing one without any disks I receive the following error messages:

```
Error: Failed adding new disk: Failed to bootstrap OSD: Failed to run: ceph-osd --mkfs --no-mon-config -i 0: exit status 250 (2023-01-25T10:57:28.939-0500 ffff8c68b020 -1 bluestore(/var/lib/ceph/osd/ceph-0) _open_db_and_around failed to load os-type: (2) No such file or directory
2023-01-25T10:57:28.939-0500 ffff8c68b020 -1 bluestore(/var/lib/ceph/osd/ceph-0) mkfs fsck found fatal error: (2) No such file or directory
2023-01-25T10:57:28.939-0500 ffff8c68b020 -1 OSD::mkfs: ObjectStore::mkfs failed with error (2) No such file or directory
2023-01-25T10:57:28.939-0500 ffff8c68b020 -1  ** ERROR: error creating empty object store in /var/lib/ceph/osd/ceph-0: (2) No such file or directory)
```

Here is a screenshot:
![image](https://user-images.githubusercontent.com/81767471/214612956-c5a2b3be-435b-4339-8f6a-f2fe11fc796b.png)

I have confirmed the location `/var/lib/ceph` does not exist.

**Comment by sabaini** [comment]
Hey James,

I can't easily repro this here (testing on Ubuntu Jammy, microceph  0+git.00fe8d8).

```
$ lsblk /dev/sdb
NAME MAJ:MIN RM SIZE RO TYPE MOUNTPOINTS
sdb    8:16   0  16G  0 disk 
$ sudo microceph disk add /dev/sdb
$ sudo microceph.ceph osd tree
ID  CLASS  WEIGHT   TYPE NAME                     STATUS  REWEIGHT  PRI-AFF
-1         0.01559  root default                                           
-2         0.01559      host devcontainer-d3654b                           
 0         0.01559          osd.0                     up   1.00000  1.00000
```

For repro purposes, can I ask which platform and microceph version you are running? Also, which command you ran to add the disk?

cheers,
peter.

**Comment by james-luther** [comment]
Thanks for the reply peter. Here's the details:

It's a Raspberry Pi 4 8GB running Ubuntu 22.04.1 Jammy with Microceph 0+git.00fe8d8. 

I'm wondering if there is a prerequisite missing beacuse deploying ceph via cephadm also doesn't allow use of any devices.

**Comment by stgraber** [comment]
I wonder if it's the `:0` in the path which is causing us to put a broken symlink in place or something when generating the OSD dir.
This would then explain the mkfs failure.

@james-luther Can you show `ls -lh /var/snap/microceph/common/data/osd/ceph-0/`? With a bit of luck we didn't clean it up on failure and there will be enough left to figure this out.

**Comment by james-luther** [comment]
It could be the :0 path. I will try with another drive to verify. Until then, this is the output:

![image](https://user-images.githubusercontent.com/81767471/214918068-429c48d2-20a7-476c-992b-23890a98c7b1.png)

**Comment by james-luther** [comment]
Ok, it appears these drives are bad. I purchased a batch of them and they just don't work correctly. I tried another spare samsung drive and everything works as intended:

![image](https://user-images.githubusercontent.com/81767471/214921670-f362a942-4687-4705-b30d-a22bc5e3335c.png)

This disk worked immediately. I'm going to do some further testing on the disks that aren't working to see if they can be fixed but because they were old stock and are causing issues, I assume there is no saving them.
